BootloaderCommonPkg/Library/TpmLib/TpmLib.c

```diff
@@ -162,26 +223,34 @@ TpmPcrBankCheck (
 
   Status = Tpm2GetCapabilitySupportedAndActivePcrs (&TpmHashAlgorithmBitmap, &ActivePcrBanks);
   if (Status == EFI_SUCCESS) {
+
+    DEBUG ((DEBUG_INFO, "TpmHashAlgorithmBitmap 0x%08x ActivePcrBanks 0x%08x\n", TpmHashAlgorithmBitmap, ActivePcrBanks));
+
     // Update Active PCR banks info in TPM Library data
     TpmLibSetActivePcrBanks (ActivePcrBanks);
 
-    // Check if TPM supports SHA256 PCR bank
-    if ((TpmHashAlgorithmBitmap & HASH_ALG_SHA256) == 0) {
-      DEBUG ((DEBUG_ERROR, "SHA256 hash algorithm is NOT supported. This should never happen with TPM2.0 device. \n"));
+    // Check if TPM supports  PCR bank
+    if (!(TpmHashAlgorithmBitmap & HASH_ALG_SHA256) &&  !(TpmHashAlgorithmBitmap & HASH_ALG_SHA384)
+                                                    && !(TpmHashAlgorithmBitmap & HASH_ALG_SM3_256)){
+      DEBUG ((DEBUG_ERROR, "Hash algorithm is NOT supported. This should never happen with TPM2.0 device. \n"));

```

Comment:
Better uses below to make it follow coding style and clear.
if ((TpmHashAlgorithmBitmap  & (HASH_ALG_SHA256 | HASH_ALG_SHA384 | HASH_ALG_SM3_256))  == 0) {

Reply:
will update.